morpheushelper/cogs/alert_channel.py

```diff
@@ -0,0 +1,127 @@
+    if alert_channel_id <= 0:
+        return None
+
+    text_channel: TextChannel = get(guild.text_channels, id=alert_channel_id)
+    if not text_channel:
+        await Settings.set(int, "alert_channel", -1)
+
+    return text_channel
+
+
+async def getMaxHops() -> int:
+    """
+    Retrieves the channel hops per minute in order for a message to appear
+    """
+    return await Settings.get(int, key="alert_channel_warn_channel_hops", default=5)
+
+
+class AlertChannelCog(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.user_hops = {}
+
+    async def on_voice_state_update(self, member: Member, before: VoiceState, after: VoiceState):
+        """
+        Checks for channel-hopping
+        """
+        if not (before.channel and after.channel):
+            return
+        try:
+            self.user_hops[member.id] += 1
+        except KeyError:
+            self.user_hops[member.id] = 1
+
+        tempMax = await getMaxHops()
+        if self.user_hops[member.id] >= tempMax > 0:
+            del self.user_hops[member.id]
+            embed = Embed(title=translations.alert_channel_hop, color=Colours.AlertChannel)
+            embed.add_field(name=translations.member, value=member.mention)
+            embed.add_field(name=translations.alert_channel_hop_current_channel, value=after.channel.name)
+            if ch := await getAlertChannel(member.guild):
+                await ch.send(embed=embed)
+            else:
+                logging.warning("No alert channel so far")
+
+    @tasks.loop(minutes=1)
+    async def hop_loop(self):
+        # Once a minute, all possible channel hops are being reset
+        self.user_hops = {}
+
+    @commands.group(name="alert")
+    @PermissionLevel.ADMINISTRATOR.check
+    async def alert_channel(self, ctx: Context):
+        """
+        Configures the alert channel
+        """
+        if not ctx.subcommand_passed:
+            embed = Embed(title=translations.alert_channel, colour=Colours.AlertChannel)
+            channel: TextChannel = await getAlertChannel(ctx.guild)
+            if channel:
+                embed.add_field(name=translations.alert_channel_get,
+                                value=channel.mention,
+                                inline=False)
+            else:
+                embed.add_field(name=translations.alert_channel_get,
+                                value=translations.none,
+                                inline=False)
+            embed.add_field(name=translations.alert_channel_hop_current_amount, value=str(await getMaxHops()),
+                            inline=False)
+            await ctx.send(embed=embed)
+
+    @alert_channel.command(name="set")
+    async def alertch_set(self, ctx: Context, channel: Union[TextChannel, int]):
+        """
+        Updated the alert channel (set channel to `0` to unset)
+        """
+        if isinstance(channel, int):
```

Comment:
it would be nice if you combine both if statements to one like this so you can save one nest.
```python
if isinstance(channel, int) and channel == 0:
    await Settings.set(int, "alert_channel", 0)
else:
    await Settings.set(int, "alert_channel", channel.id)

# ...
embed.add_field(name=translations.channel, value=str(channel) if channel else translations.none)
# ...
```